qadence/ml_tools/config.py

```diff
@@ -60,3 +71,262 @@ def __post_init__(self) -> None:
+    """
+    Range of data the data encoder assumes as natural range.
+
+    Give a single tuple to use the same range for all features.
+    Give a dict of (str, tuple) where the key is the name of the variable and the
+    value is the target range to use for that feature.
+    """
+
+    multivariate_strategy: str = "parallel"
+    """
+    The  encoding strategy in case of multi-variate function.
+
+    If "parallel",
+    the features are encoded in one block of rotation gates. with each
+    feature given an equal number of qubits. If "serial", the features are
+    encoded sequentially, with an ansatz block between. "parallel" is allowed
+    only for "digital" `feature_map_strategy`.
+    """
+
+    feature_map_strategy: str = "digital"
+    """Strategy for feature map.
+
+    Accepts 'digital', 'analog' or 'rydberg'. Defaults to "digital".
+    If the strategy is incompatible with the `operation` chosen, then `operation`
+    gets preference and the given strategy is ignored.
+    """
+
+    param_prefix: str | None = None
+    """
+    String prefix to create trainable parameters in Feature Map.
+
+    A string prefix to create trainable parameters multiplying the feature parameter
+    inside the feature-encoding function. Note that currently this does not take into
+    account the domain of the feature-encoding function.
+    Defaults to `None` and thus, the feature map is not trainable.
+    Note that this is separate from the name of the parameter.
+    The user can provide a single prefix for all features, and they will be appended
+    by appropriate feature name automatically.
+    """
+
+    num_repeats: int | dict[str, int] = 0
+    """
+    Number of feature map layers repeated in the data reuploadig step.
+
+    If all are to be repeated the same number of times, then can give a single
+    `int`. For different number of repeatitions for each feature, provide a dict
+    of (str, int) where the key is the name of the variable and the value is the
+    number of repeatitions for that feature.
+    This amounts to the number of additional reuploads. So if `num_repeats` is N,
+    the data gets uploaded N+1 times. Defaults to no repeatition.
+    """
+
+    operation: Callable[[Parameter | Basic], AnalogBlock] | Type[RX] | None = None
+    """
+    Type of operation.
+
+    Choose among the analog or digital rotations or a custom
+    callable function returning an AnalogBlock instance. If the type of operation is
+    incompatible with the `strategy` chosen, then `operation` gets preference and
+    the given strategy is ignored.
+    """
+
+    inputs: list[Basic | str] | None = None
+    """
+    List that indicates the order of variables of the tensors that are passed.
+
+    Optional if a single feature is being encoded, required otherwise. Given input tensors
+    `xs = torch.rand(batch_size, input_size:=2)` a QNN with `inputs=("t", "x")` will
+    assign `t, x = xs[:,0], xs[:,1]`.
+    """
```

Comment:
Shouldn't the type of `inputs` be a tuple, rather than a list? This would agree with the types of most other attributes of this config. Also, the docstring here also hints to define `inputs` as a tuple.

Reply:
I would be in favor of correcting the docstrings instead. See [this](https://github.com/pasqal-io/qadence/blob/ds/remove_tm/qadence/model.py#L100) or any other such time in `QNN` or `QuantumModel` where the `inputs` are in a list, so just trying to keep it consistent with that.

Reply:
I see, and it would be out of the scope of this PR to do a refactoring at that level. Still, QNN inputs and such look like hashable things to me. Maybe it should be considered for the future? @dominikandreasseitz